src/pages/library/SearchAndFilterComponent.ts

```diff
@@ -294,7 +294,7 @@ const FilterSectionComponent = <ListItem>({ section, fullList, selectedSections,
   const valuesToShow = _.filter(sectionEntry => _.size(listItemsMatchForSectionEntry(sectionEntry, section.matchBy, fullList)) > 0, section.values)
 
   return h(Fragment, [
-    _.map(sectionEntry => {
+    _.map((sectionEntry : string) => {
```

Comment:
Not sure why this was necessary, it would be nice to avoid this if possible.

Reply:
Here's what I don't get. If I assign the result of `take()` to an untyped const, like this:
```
  const truncatedValues = _.take(numLabelsToRender, valuesToShow)
```
Then typescript can infer the type for `truncatedValues` correctly. If I use it inline in the call to `map()` then it can't. It looked I need to either type the lambda argument, cast the result of the `take()` call, or add a local `const` that's only used in one place. Of all these solutions I like the `const` the best since it doesn't require adding explicit types to the code. Avoiding types is better IMO because less code needs to change if our types ever change.

Reply:
I agree with that summary.

Reply:
I'm going to write up a short example and post to #terra-ui to follow up on this.

Reply:
I think these issues are really important to capture. When we were first introducing TS, these types of issues were part of my intuitive flags about a TS conversion. But some of these issues are so esoteric, when not working with it everyday it becomes hard to articulate some of the things you run into.

Do you think it would be worth having a "TS miscellany" doc we could point to?